examples/slice/slice_mountain_paradiag.py

```diff
@@ -0,0 +1,222 @@
+import firedrake as fd
+import asQ
+import numpy as np
+from slice_utils import hydrostatic_rho, \
+    get_form_mass, get_form_function, maximum
+from petsc4py import PETSc
+dT = fd.Constant(1)
+
+nlayers = 50  # horizontal layers
+base_columns = 150  # number of columns
+L = 144e3
+distribution_parameters = {"partition": True, "overlap_type": (fd.DistributedMeshOverlapType.VERTEX, 2)}
+
+# set up the ensemble communicator for space-time parallelism
+nspatial_domains = 4
+M = [2, 2, 2, 2]
+
+# we expect 16 processors
+
+ensemble = fd.Ensemble(fd.COMM_WORLD, nspatial_domains)
```

Comment:
Can we use the `asQ.create_ensemble` factory here? The slice distribution is hard-coded for the time being but it means at least the number of spatial domains will adapt automatically